Lesson3/Task3/Model.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.IO;
+using System.IO.Compression;
+
+namespace Task3
+{
+    class Model
+    {
+        public string FilePath { get; private set; }
+        public bool IsPath { get; private set; }
+
+        public string CompressFile()
+        {
+            FileStream source = new FileStream(FilePath, FileMode.Open);
+            FileStream destination = File.Create(Path.GetDirectoryName(FilePath) + @"\archive.zip");
+
+            GZipStream compressor = new GZipStream(destination, CompressionMode.Compress);
+
+            int inputByte = source.ReadByte();
+            while (inputByte != -1)
+            {
+                compressor.WriteByte((byte)inputByte);
+                inputByte = source.ReadByte();
+
+            }
+
+            compressor.Close();
+
+            return destination.Name;
+        }
+
+        public string ShowFileContent()
+        {
+            StreamReader streamReader = File.OpenText(FilePath);
+            string content = streamReader.ReadToEnd();
+            streamReader.Close();
+
+            return content;
+        }
+
+        public bool SearchFile(string targetDirectory, string fileName)
+        {
+            DirectoryInfo dir = new DirectoryInfo(targetDirectory);
+            FileInfo[] fileInfo;
+
+            if (!dir.Exists)
+            {
+                return false;
+            }
+
+            fileInfo = dir.GetFiles(fileName);
+
+            if (fileInfo.Length == 0)
+            {
+                foreach (var directory in dir.GetDirectories())
+                {
+                    if (directory.Attributes == (FileAttributes.System | FileAttributes.Hidden | FileAttributes.Directory))
```

Comment:
if you want to check if `directory.Attributes` contains one of the flags you need to do it in another way:
```
var flags = BindingFlags.Instance | BindingFlags.Static;

Console.WriteLine(flags == (BindingFlags.Instance | BindingFlags.Static)); // true
Console.WriteLine(flags == BindingFlags.Instance); // false
         
Console.WriteLine(new string('-', 40));
Console.WriteLine((flags & BindingFlags.Instance) != 0); // true
Console.WriteLine((flags & BindingFlags.Static) != 0); // true
Console.WriteLine((flags & (BindingFlags.Static | BindingFlags.Instance)) != 0); // true
Console.WriteLine((flags & BindingFlags.DeclaredOnly) != 0); // false

Console.WriteLine(new string('-', 40));
Console.WriteLine(flags.HasFlag(BindingFlags.Instance)); // true
Console.WriteLine(flags.HasFlag(BindingFlags.Static)); // true
Console.WriteLine(flags.HasFlag(BindingFlags.Instance | BindingFlags.Static)); // true
Console.WriteLine(flags.HasFlag(BindingFlags.DeclaredOnly)); // false
```